custom_components/google_home/api.py

```diff
@@ -309,6 +325,49 @@ async def reboot_google_device(self, device: GoogleHomeDevice) -> None:
                 device.name,
             )
 
+    async def get_or_set_do_not_disturb(
+        self, device: GoogleHomeDevice, enable: Optional[bool] = None
+    ) -> GoogleHomeDevice:
+        """Gets or sets the do not disturb setting on a Google Home device."""
+
+        data = None
+
+        if enable is None:
+            _LOGGER.debug(
+                "Getting Do Not Disturb setting from Google Home device %s",
+                device.name,
+            )
+        else:
+            data = {"notifications_enabled": not enable}
+            _LOGGER.debug(
+                "Setting Do Not Disturb setting to %s on Google Home device %s",
+                not enable,
+                device.name,
+            )
+
+        response = await self.post(
+            endpoint=API_ENDPOINT_DO_NOT_DISTURB, data=json.dumps(data), device=device
+        )
+        if response:
+            if "notifications_enabled" in response:
+                is_enabled = bool(response["notifications_enabled"])
+                _LOGGER.debug(
+                    "Received Do Not Disturb setting from Google Home device %s"
+                    " - Enabled: %s",
+                    device.name,
+                    not is_enabled,
+                )
+
+                device.set_do_not_disturb(is_enabled)
```

Comment:
Maybe you have some confusion here? Shouldn't it be?
```python
device.set_do_not_disturb(not is_enabled)
```

Reply:
This may explain why you have blinking state.

Reply:
Yes, that was why i was confused about what you where talking about above. Saw this and have changed it now. 😃

Reply:
Rename `is_enabled` to `notifications_enabled` to avoid confusion.